src/api/endpoints/auth/TokenManager.ts

```diff
@@ -30,11 +32,12 @@ export default class TokenManager extends ApiEndpoint {
 
   public async getAccessToken(source: string = "unknown"): Promise<string | undefined> {
     this.log?.debug(`getAccessToken: New token requested from ${source}`);
-    if (this.accessToken !== undefined) {
-      const exp = TokenManager.getJwtFromAccessToken(this.accessToken).exp;
+    if (this.accessToken !== undefined && this.accessTokenExpiration !== undefined) {
+      const exp = Math.trunc(new Date(this.accessTokenExpiration).getTime() / 1000);
```

Comment:
Seems like we could just store this value when we fetch a new access token instead of recalculating it every time we use said token? Or are there threading reasons we can't?

Reply:
I put it there to not lose the milliseconds we discard in case we need to use them in the futur but we can always do what you're suggesting and change it back in the futur if we need too, what do you think?

Reply:
No big deal either way, it's cheap relative to the cost of a web request

Reply:
I'll keep that way for now